### ISSUE TITLE
a11y CRITICAL: `auro-menu`: no label on `role='listbox'` element

### ISSUE THREAD

**Issue by sun-mota** [AlaskaAirlines/auro-formkit#595]
### What browser client(s) are you seeing the issue on?

- [ ] Firefox
- [ ] Chrome
- [ ] Safari
- [ ] Microsoft Edge

### What devices platform(s) are you seeing the issue on?

- [ ] Desktop
- [ ] Mobile

### Steps to reproduce

 Rule: aria-input-field-name
      Impact: serious
      ARIA input fields must have an accessible name (https://dequeuniversity.com/rules/axe/4.10/aria-input-field-name?application=axeAPI)
      
      Issue target: auro-menu
      Context: <auro-menu matchword="o" role="listbox" root="" aria-busy="false" disabled="">
                <auro-menuoption value="option 1" event="uniqueEventName" tabindex="-1" role="option" aria-selected="false"><strong>o</strong>pti<strong>o</strong>n 1</auro-menuoption>
              </auro-menu>
      Fix any of the following:
        aria-label attribute does not exist or is empty
        aria-labelledby attribute does not exist, references elements that do not exist or references elements that are empty
        Element has no title attribute

### Expected behavior

`role=listbox` element should have either `aria-labelledby` or `aria-label`

### Reproducible on the Auro doc site

- [ ] I am able to reproduce this issue on the [Auro doc site](https://auro.alaskaair.com).

**Comment by blackfalcon** [comment]
Adding @blackfalcon as a participant

See parent issue

- https://github.com/AlaskaAirlines/auro-formkit/issues/497
